.github/workflows/ci.yml

```diff
@@ -11,29 +11,29 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Set up Python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
 
       - name: Run pre-commit
-        run: pipx run -- hatch run lint:run
+        run: python3 -m pip install hatch && hatch run lint:run
```

Comment:
**n.b.** Due to `pipx run --` usage, the cpython version being used was always `python 3.10` despite what was set by `actions/setup-python@v5`.